module/mempool/stdmap/incorporated_result_seals.go

```diff
@@ -46,15 +129,66 @@ func (ir *IncorporatedResultSeals) ByID(id flow.Identifier) (*flow.IncorporatedR
 
 // Rem removes an IncorporatedResultSeal from the mempool
 func (ir *IncorporatedResultSeals) Rem(id flow.Identifier) bool {
-	return ir.Backend.Rem(id)
+	removed := false
+	err := ir.Backend.Run(func(entities map[flow.Identifier]flow.Entity) error {
+		var entity flow.Entity
+		entity, removed = ir.Backdata.Rem(id)
+		if !removed {
+			return nil
+		}
+		seal := entity.(*flow.IncorporatedResultSeal)
+		ir.removeFromIndex(id, seal.Header.Height)
+		return nil
+	})
+	if err != nil {
+		panic(err)
+	}
+	return removed
 }
 
-// Clear removes all entities from the pool.
 func (ir *IncorporatedResultSeals) Clear() {
-	ir.Backend.Clear()
+	err := ir.Backend.Run(func(_ map[flow.Identifier]flow.Entity) error {
+		ir.Backdata.Clear()
+		ir.byHeight = make(map[uint64]sealSet)
```

Comment:
```suggestion
		ir.Backdata.Clear()
		ir.lowestHeight = 0
		ir.byHeight = make(map[uint64]sealSet)
```
Not sure if this is actually used, but I would expect this to be reset as well here

Reply:
The Clear method is only used when there is execution fork detected. In that case, we would like to remove all data, but no the pruned height, because in case another call attempts to prune again, the lowest height is there be used for checking